client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java

```diff
@@ -213,6 +217,10 @@ public long pushData() throws IOException {
 
       long freedBytes = freeMemory();
       inMemSorter.freeMemory();
+      // Reset the in-memory sorter's pointer array only after freeing up the memory pages holding
+      // the records. Otherwise, if the task is over allocated memory, then without freeing the
+      // memory pages, we might not be able to get memory for the pointer array.
+      taskContext.taskMetrics().incMemoryBytesSpilled(freedBytes);
```

Comment:
I think we need to inc the metric for `cleanupResources` too.

Reply:
Yea, done